### ISSUE TITLE
solution

### ISSUE THREAD

**Issue by sbenebrh** [udacity/ud120-projects#172]
can i have the solution of the mini-project naive bayes please

**Comment by AdityaVallabh** [comment]
I hope this helps:
```python
from sklearn.naive_bayes import GaussianNB
from sklearn.metrics import accuracy_score
clf = GaussianNB()

t0 = time()
clf.fit(features_train, labels_train)
print("Training Time:" + str(round(time()-t0, 3)) + "s")

t0 = time()
pred = clf.predict(features_test)
print("Predicting Time:" + str(round(time()-t0, 3)) + "s")

accuracy = accuracy_score(labels_test, pred)
print('Accuracy: ' + str(accuracy))
```

**Comment by Onestroke1** [comment]
My accuracy was 0.9203640500568828 but the Udacity quiz said it was expecting a much high accuracy. My code looks just like yours. I'm wondering why my is not right?

**Comment by xkludge** [comment]
Can you paste your code?

**Comment by Onestroke1** [comment]
from time import time
from email_preprocess import preprocess
from sklearn.naive_bayes import GaussianNB
from sklearn.metrics import accuracy_score

features_train, features_test, labels_train, labels_test = preprocess()

clf = GaussianNB()

t0 = time()
clf.fit(features_train, labels_train)
print ("training time:", round(time()-t0, 3), "s")

t0 = time()
pred = clf.predict(features_test)
print ("prediction time:", round(time()-t0, 3), "s")
accuracy = accuracy_score(pred, labels_test)

print("Accuracy", accuracy)

**Comment by xkludge** [comment]
You have your params reversed on accuracy.

`accuracy_score(true_label, predicted_labels)`

https://scikit-learn.org/stable/modules/generated/sklearn.metrics.accuracy_score.html

**Comment by Onestroke1** [comment]
I reversed the parameters as suggested and got the same accuracy as before.

**Comment by xkludge** [comment]
Instead of using `accuracy_score` you could use the `clf.score` method.

**Comment by Onestroke1** [comment]
> Instead of using `accuracy_score` you could use the `clf.score` method.

That produces an error. Not sure how to implement that.

**Comment by xkludge** [comment]
It was a while ago I did this but from my notes it looked like this:

```python
import numpy as np
from sklearn.naive_bayes import GaussianNB

clf = GaussianNB()
t0 = time()
clf.fit(features_train, labels_train)
print "training time:", round(time()-t0, 3), "s"

t1 = time()
print("accuracy: {} ".format(clf.score(features_test, labels_test)))
print "predicting time:", round(time()-t1, 3), "s"
```
